File: twopointer/392-is-subsequence/IsSubsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if len(s)==0:
            return True
        if len(t)==0:
            return False
        tr = 0
        for ch in t:
            if tr<len(s) and s[tr]==ch:
                tr+=1
        return tr == len(s)

File: twopointer/392-is-subsequence/IsSubsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        if(s.length()==0){
            return true;
        }
        if(s.length()==0){
            return false;
        }
        int tr = 0;
        for(int i=0;i<t.length();i++){
            if(s.charAt(tr)==t.charAt(i)){
                tr++;
                if(tr==s.length()){
                    return true;
                }
            }
            
        }
        return false;
    }
}